Nodejs/Setup/Custom_Action/CustomAction.cs

```diff
@@ -0,0 +1,191 @@
+                  {
+                      {"baseType","EventData"},
+                      {"baseData",new Dictionary<string,object>()
+                      {
+                          {"ver","2"},
+                          {"name","NTVSInstallerTel"},
+                          {"properties",new Dictionary<string,object>()
+                          {
+                               {"InstallStatus","Success"},
+                               {"IsNTVSInstalled",isInstalled.ToString()},
+                               {"CurrentState",currentState},
+                               {"RequestState",requestState},
+                               {"NTVSVersion",session["NTVSVersion"]},
+                               {"VSVersion",session["VSVersion"]},
+                               {"MSIVersion",session["MSIVersion"]},
+                               {"TimeTakenInSeconds",installTime.TotalSeconds.ToString()}
+                          }}
+                      }}
+                  }}
+            };
+
+            string jsonString = (new JavaScriptSerializer()).Serialize(data);          
+            using (WebClient client = new WebClient())
+            {
+                string response =     client.UploadString("https://dc.services.visualstudio.com/v2/track", jsonString);
+                session.Log(response);
+            }
+            session.Log("End Telemetry Log");
+            return ActionResult.Success;
+        }
+
+        public static ActionResult LogInstallErrorResult(Session session)
+        {
+            TimeSpan installTime = DateTime.Now - DateTime.Parse(session["InstallStartTime"]);
+
+            session.Log("Begin Telemetry Log");
+            bool isInstalled = session.EvaluateCondition("Installed");
+            string currentState = "";
+            string requestState = "";
+            session.Log("Is Aready Installed: " + session.EvaluateCondition("Installed").ToString());
+
+            FeatureInfoCollection fc = session.Features;
+            foreach (FeatureInfo f in fc)
+            {
+                currentState = f.CurrentState.ToString();
+                requestState = f.RequestState.ToString();
+                break; // we just want the current and requested state of A feature to understand if its a new user, upgrade, reinstall or remove.
+            }
+
+            session.Log("Starting POST");
+            Dictionary<string, object> data = new Dictionary<string, object>()
+            {
+                { "iKey", "377a3718-78a7-49df-abcc-1001317db729" },
+                 { "name", "Microsoft.ApplicationInsights.Event"},
+                {"time", DateTime.Now.ToUniversalTime().ToString()},
+                  {"data",new Dictionary<string,object>()
+                  {
+                      {"baseType","EventData"},
+                      {"baseData",new Dictionary<string,object>()
+                      {
+                          {"ver","2"},

```

Comment:
What's this version number and why is it always "2"? Seems worth having a constant defined.